lib/imsg-buffer.c

```diff
@@ -209,8 +209,10 @@ msgbuf_clear(struct msgbuf *msgbuf)
 {
 	struct ibuf	*buf;
 
-	while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL)
+	while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL) {
 		ibuf_dequeue(msgbuf, buf);
+		TAILQ_FIRST(&msgbuf->bufs) = NULL;
```

Comment:
I am pretty sure that this is not the right thing to do, since we set bufs to NULL prematurely.  Suppose msgbuf-bufs has 5 entries, we dequeue the first one and then set the msgbuf-bufs to NULL and start leaking memory.

I think this is the right thing to do here:
diff --git a/lib/imsg-buffer.c b/lib/imsg-buffer.c
index a486fc1..9ac63ff 100644
--- a/lib/imsg-buffer.c
+++ b/lib/imsg-buffer.c
@@ -207,10 +207,12 @@ msgbuf_drain(struct msgbuf *msgbuf, size_t n)
 void
 msgbuf_clear(struct msgbuf *msgbuf)
 {
-       struct ibuf     *buf;
+       struct ibuf     *buf, *next;
 
-       while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL)
+       for (buf = TAILQ_FIRST(&msgbuf->bufs); buf != NULL; buf = next) {
+               next = TAILQ_NEXT(buf, entry);
                ibuf_dequeue(msgbuf, buf);
+       }
 }
 
 int